frontend/src/Services/LocalStorageService.ts

```diff
@@ -13,5 +13,42 @@ class LocalStorageService {
     localStorage.setItem("Session", JSON.stringify(newArray));
     return newArray;
   };
+  public static GetSelectSession = (
+    sessionId: string
+  ): SessionState | undefined => {
+    const storageArray = JSON.parse(localStorage.getItem("Session")!);
+    return storageArray?.find(
+      (session: { id: string }) => session.id === sessionId
+    );
+  };
+  public static SetStorageArray = (input: SessionState) => {
+    const storageArray = LocalStorageService.GetSessions();
+    if (storageArray) {
+      const x = storageArray.findIndex(
+        (session: SessionState) => session.id === input.id
+      );
+      if (x < 0) {
+        localStorage.setItem(
+          "Session",
+          JSON.stringify([...storageArray, input])
+        );
+      } else {
+        storageArray[x] = input;
+        localStorage.setItem("Session", JSON.stringify(storageArray));
+      }
+    } else {
+      localStorage.setItem("Session", JSON.stringify([input]));
+    }
+  };
+  public static GetFromStorage = (): SessionState | undefined => {
+    const getLocalState = localStorage.getItem("Session");
+    if (!getLocalState) {
+      return undefined;
+    }
+    const localState = JSON.parse(getLocalState);
+    return {
+      ...localState,
+    } as SessionState;
```

Comment:
I might be misremembering, but isn't the contents of the item "Session" in localstorage an ARRAY of SessionStates?

So... why are we then spitting out all the entries of the array into an object and returning it?
`localState` in this segment should technically be a type Array<SessionState> and the output of this function should be `Array<SessionState> | []`

I think this function is a hold-over from before we changed the localstorage into being an array of SessionStates.
Given the changes we've made, you should test and see if we still need the @@INIT reducer at all, and if so we need to then update the logic to interact correctly with our new localstorage structure